yjit/src/codegen.rs

```diff
@@ -646,12 +640,12 @@ pub fn gen_entry_prologue(cb: &mut CodeBlock, iseq: IseqPtr, insn_idx: u16) -> O
     // compiled for is the same PC that the interpreter wants us to run with.
     // If they don't match, then we'll take a side exit.
     if unsafe { get_iseq_flags_has_opt(iseq) } {
-        gen_pc_guard(&mut asm, iseq, insn_idx);
+        gen_entry_guard(&mut asm, ocb, iseq, insn_idx);
     }
 
     asm.compile(cb);
 
-    if cb.has_dropped_bytes() {
+    if cb.has_dropped_bytes() || ocb.unwrap().has_dropped_bytes() {
```

Comment:
Not directly related, we should probably make `Assembler::compile` return a `Result` soon. The dropped bytes flag being returned implicitly is pretty unnecessary, especially since `compile()` internally checks that flag! I think bubbling with `?` is much better.

Reply:
I'll scope out the `asm.compile` part from this PR, but I reverted this `|| ocb.unwrap().has_dropped_bytes()` by using `Option` and `?` https://github.com/ruby/ruby/pull/7535/commits/73e017a92c9dc11128d1b734053a2bbbffe48095.